docs/design/proposals/rbd-mirror.md

```diff
@@ -0,0 +1,96 @@
+
+## Architecture
+
+![mirror](rbd-mirror.png)
+
+## Design
+
+Currently, CEPH-CSI generates its unique ID for each RBD image and stores the
+corresponding PersistentVolume (PV) name and the unique ID mapping, It creates
+the RBD image with the unique ID and returns the encoded value which contains
+all the required information for other operations. For mirroring, the same RBD
+image will be mirrored to the secondary cluster. As the journal(OMAP data) is
+not mirrored to the secondary cluster, The RBD images corresponding to the PV
+can not be identified without OMAP data.
+
+**Pre-req** It's expected that the Kubernetes Admin/User will create the static
+PersistentVolumeClaim (PVC) on the secondary site during the failover.
+
+**Note:** when the static PVC created on the secondary site we cannot use the
+VolumeHandle to identify the OMAP data nor the image anymore because as we have
+only PoolID and ClusterID in the VolumeHandle. we cannot identify the correct
+pool name from the PoolID because pool name will remain the same on both
+clusters but not the PoolID even the ClusterID can be different on the
+secondary cluster.
+
+> Sample PV spec which will be used by controller to regenerate OMAP data
+
+```yaml
+csi:
+      controllerExpandSecretRef:
+        name: rook-csi-rbd-provisioner
+        namespace: rook-ceph
+      driver: rook-ceph.rbd.csi.ceph.com
+      fsType: ext4
+      nodeStageSecretRef:
+        name: rook-csi-rbd-node
+        namespace: rook-ceph
+      volumeAttributes:
+        clusterID: rook-ceph
+        imageFeatures: layering
+        imageFormat: "2"
+        imageName: csi-vol-0c23de1c-18fb-11eb-a903-0242ac110005
+        journalPool: replicapool
+        pool: replicapool
+        radosNamespace: ""
+      volumeHandle: 0001-0009-rook-ceph-0000000000000002-0c23de1c-18fb-11eb-a903-0242ac110005
+```
+
+> **VolumeHandle** is the unique volume name returned by the CSI volume plugin’s
+CreateVolume to refer to the volume on all subsequent calls.
+
+Once the static PVC is created on the secondary cluster, the Kubernetes User
+can try delete the PVC,expand the PVC or mount the PVC. Incase of mounting
+(NodeStageVolume) we will get the volume context in RPC call but not in the
+Delete/Expand Request. In Delete/Expand RPC request only the VolumeHandle
+(`clusterID-poolID-volumeuniqueID`) will be sent where it contains the encoded
+information related to ClusterID and PoolID. The VolumeHandle is not useful in
+secondary cluster as the PoolID and ClusterID always may not be the same.
+
+To solve this problem, We will have a new controller running as part of
+provisioner pod which watchers for the PV objects. When a PV is created it will
+extract the required information from the PV spec and it will regenerate the
+OMAP data and also it will generate a new VolumeHandle
+(`newclusterID-newpoolID-volumeuniqueID`) and it creates an OMAP object for
+mapping between old VolumeHandle and new VolumeHandle. whenever cephcsi gets a
```

Comment:
*cephcsi* -> *Ceph-CSI* , please use the same spelling/capitalization everywhere